dask/compatibility.py

```diff
@@ -56,10 +60,26 @@ def _getargspec(func):
     range = xrange
     reduce = reduce
     operator_div = operator.div
+    FileNotFoundError = IOError
 
     def _getargspec(func):
         return inspect.getargspec(func)
 
+    def gzip_decompress(b):
+        f = gzip.GzipFile(fileobj=BytesIO(b))
+        result = f.read()
```

Comment:
gzip maintains a bit of a buffer that will need to be flushed out.  It also has a few termination bytes.

Compression formats are complex.  Closing is good practice.